packages/google_maps_flutter/google_maps_flutter/lib/src/google_map.dart

```diff
@@ -57,6 +61,13 @@ class GoogleMap extends StatefulWidget {
   })  : assert(initialCameraPosition != null),
         super(key: key);
 
+  /// Restoration ID to save and restore state of the native widget.
+  ///
+  /// If multiple [GoogleMap] widgets are included in the same layout, you must
+  /// specify a unique value for each widget in order to ensure native state is
+  /// restored correctly. Otherwise, the argument is optional.
```

Comment:
In the framework we usually only enable state restoration if an ID has been provided.

Reply:
I could go that route, but the native Android component generally assumes native state is being saved correctly so I'd prefer to save state proactively.